.github/workflows/publish-charm.yaml

```diff
@@ -0,0 +1,26 @@
+name: publish-charm
+
+on:
+  workflow_call:
+
+jobs:
+  publish-charm:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install charmcraft
+        run: sudo snap install charmcraft --classic
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Move charm in current directory
+        run: find ./ -name sdcore-amf_ubuntu-22.04-amd64.charm -exec mv -t ./ {} \;
+      - name: Select Charmhub channel
+        uses: canonical/charming-actions/channel@2.2.5
+        id: channel
+      - name: Upload charm to Charmhub
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
```

Comment:
This won't work as it is, you can look at this project's [publish.yaml](https://github.com/canonical/self-signed-certificates-operator/blob/main/.github/workflows/publish-charm.yaml) and [push.yaml](https://github.com/canonical/self-signed-certificates-operator/blob/main/.github/workflows/push.yaml).

Reply:
I added the secrets section like at the beginning of the example you provided, otherwise they look pretty much the same.
Let me know if this was the intention :)